cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -38,13 +38,13 @@
 
 ;; x*0 == 0*x == x.
 (rule (simplify (imul ty
-                        x
-                        (iconst ty (u64_from_imm64 0))))
-      (iconst ty (imm64 0)))
+                      _
+                      zero @ (iconst ty (u64_from_imm64 0))))
+      (subsume zero))
 (rule (simplify (imul ty
-                        (iconst ty (u64_from_imm64 0))
-                        x))
-      (iconst ty (imm64 0)))
+                      zero @ (iconst ty (u64_from_imm64 0))
+                      _))
+      (subsume zero))
```

Comment:
Drive-by question not related to this actual PR, is this rule necessary given [this rule](https://github.com/bytecodealliance/wasmtime/blob/284fec127a3ea812deb3eabab09bffce3d7edd75/cranelift/codegen/src/opts/cprop.isle#L99-L101) that moves constants to the right?

Reply:
Great question. I believe you're right that this rule is not necessary since we have that one. I'm not going to delete it right now though while we're still thinking through how best to express rules involving commutative operators. Chris, Nick, Trevor, and I had a long chat on the subject earlier today, but we don't have a complete plan yet.